### PR TITLE
Folio 2187

### PR DESCRIPTION
**Pull request by ihardy**
The commit history is a bit of a mess because some things needed to be pushed to test. But this does the following:

* Modifies the `module-kubernetes` role to accept the `folio_modules` variable instead of a list of IDs.
* Adds `folio_install_type` variable (single_server or kubernetes) to separate out tasks that are specific to those environments
* modifies `common`, `yarn`, and `nodejs` roles to allow to be run on kubernetes builds.

### CODE REVIEW

File: roles/common/defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ---
+# choose single_server or kubernetes
+folio_install_type: single_server
+
 folio_user: folio
 folio_group: folio
+folio_install_type: single_server
```
Comment (wafschneider) [inline roles/common/defaults/main.yml]: This line is redundant, isn't it?

**Comment by wafschneider** [comment]
If all the tasks in the `common` role are _only_ for single server (which it looks like they are), would it be better to change how the dependency is resolved in the roles that have dependencies, rather than updating the common role so that there are many skipped tasks for a K8s install? E.g.:
```yaml
---
dependencies:
  - role: common
    when: folio_install_type == "single_server"
```

File: roles/kube-enable/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- name: enable modules
+  uri:
+    url: https://okapi-demo2.ci.folio.org/_/proxy/tenants/diku/modules
```
Comment (wafschneider) [inline roles/kube-enable/tasks/main.yml]: Shouldn't this be a variable? Maybe using `{{ okapi_url }}`?

Comment (ihardy) [reply roles/kube-enable/tasks/main.yml]: removed kube-enable in favor of using existing okapi-tenant-enable role

**Comment by ihardy** [comment]
>If all the tasks in the common role are only for single server (which it looks like they are), would it be better to change how the dependency is resolved in the roles that have dependencies, rather than updating the common role so that there are many skipped tasks for a K8s install? E.g.:

Thanks Wayne, that makes sense to me, I removed the check from the tasks in common and added it where the common role is invoked instead. 

>Same issue with the nodejs role
The nodjes role uses `folio_install_type` to determine whether to install node from apt or apk, so leaving that one as is.

**Pull request merged**